cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/ZooKeeperReconciler.java

```diff
@@ -432,11 +437,21 @@ protected Future<Void> certificateSecret(Clock clock) {
                             .reconcile(reconciliation, reconciliation.namespace(), KafkaResources.zookeeperSecretName(reconciliation.name()),
                                     zk.generateCertificatesSecret(clusterCa, Util.isMaintenanceTimeWindowsSatisfied(reconciliation, maintenanceWindows, clock.instant())))
                             .compose(patchResult -> {
-                                if (patchResult instanceof ReconcileResult.Patched) {
-                                    // The secret is patched and some changes to the existing certificates actually occurred
-                                    existingCertsChanged = ModelUtils.doExistingCertificatesDiffer(oldSecret, patchResult.resource());
-                                } else {
+                                if (patchResult != null) {
                                     existingCertsChanged = false;
+                                    if (patchResult instanceof ReconcileResult.Patched) {
+                                        // The secret is patched and some changes to the existing certificates actually occurred
+                                        existingCertsChanged = ModelUtils.doExistingCertificatesDiffer(oldSecret, patchResult.resource());
+                                    }
+                                    IntStream.range(0, zk.getReplicas())
+                                            .forEach(podNum -> {
+                                                var podName = KafkaResources.zookeeperPodName(reconciliation.name(), podNum);
+                                                zkCertificateHash.put(
+                                                        podNum,
+                                                        CertUtils.getCertificateThumbprint(patchResult.resource(),
+                                                                Ca.secretEntryNameForPod(podName, Ca.SecretEntry.CRT)
+                                                        ));
+                                            });
```

Comment:
not something to change if you like this way but quite often I ask myself why an easy old school `for` loop isn't much better and readable than "complicated" Java Streams API logic (i.e. generating a stream of int instead of just using the `for` index to go). Answer to myself is that `for` loop always wins.

Reply:
Probably a personal preference for this small loop. Personally, I'm trying to use more functional style code, and when I'm in that mindset, my mind subconsciously never picks up a for loop specifically as a valid code construct. If was reviewing this PR and it was using a for loop, I probably wouldn't have noticed and would've approved it, it's just my mental state when I'm writing code lately.

Reply:
I can totally understand the mindset of functional programming style and I totally agree when you have collections you want to work on by using streams API ... but generating a streams of integer instead of a for loop ... is too much to me ... it's diabolic :-P

Reply:
I also think a plain `for` statement would be clearer here. While there's nothing wrong _per se_ with preferring a functional style, if you're constructing a Stream simply to use `forEach` that's not functional (`forEach` is operating by side-effect, which is the opposite of functional).

@devguyio changing the code you _write_ to work around not seeing `for` as a valid code construct is not really a solution, since you still have to _read_ code written by others.

Reply:
I think my comment was misunderstood, I'm totally not saying this is better than the other, I just tried to explain the "faulty" thought process that lead to this loop. 

I changed this and the couple of ones in the tests.

Reply:
You happy with the change @tombentley?